db/db_iter.cc

```diff
@@ -390,6 +400,12 @@ void DBIter::FindNextUserEntryInternal(bool skipping, bool prefix_check) {
       break;
     }
 
+    if (TooManyInternalKeysSkipped()) {
```

Comment:
Moved counter-incrementing to be inside the function. 
I have not added statistics counters in this PR, but I'll look at adding them in a separate PR.